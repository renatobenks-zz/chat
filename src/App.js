// @flow

import * as React from 'react';
import { hot } from 'react-hot-loader';
import { Helmet } from 'react-helmet';
import { injectGlobal, ThemeProvider } from 'styled-components';

import AppRouter from './AppRouter';

import theme from './theme';

if (process.env.NODE_ENV !== 'development') {
  for (const method of [
    'assert',
    'clear',
    'count',
    'debug',
    'dir',
    'dirxml',
    'error',
    'exception',
    'group',
    'groupCollapsed',
    'groupEnd',
    'info',
    'log',
    'profile',
    'profileEnd',
    'table',
    'time',
    'timeEnd',
    'timeStamp',
    'trace',
    'warn',
  ]) {
    // $FlowFixMe
    console[method] = () => {};
  }
}

injectGlobal`
  html,
  body,
  #root {
    position: relative;
    width: 100%;
    height: 100%;
    margin: 0;
    padding: 0;
    display: flex;
    font-family: 'Montserrat', sans-serif;
    font-size: 16px;
    -webkit-font-smoothing: antialiased;
    -moz-osx-font-smoothing: grayscale;
  }

  *,
  *::before,
  *::after {
    box-sizing: border-box;
    outline: none;
  }
`;

type Props = {
  children: React.Element<typeof AppRouter>,
};

class App extends React.Component<Props> {
  render() {
    const { children } = this.props;

    return (
      <React.Fragment>
        <Helmet>
          <title>Team chat for Creditas</title>
          <link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Montserrat:200,300,400,500,600" />
        </Helmet>
        <ThemeProvider theme={theme}>{children}</ThemeProvider>
      </React.Fragment>
    );
  }
}

export default hot(module)(App);
