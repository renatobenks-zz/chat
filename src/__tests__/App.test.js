import React from 'react';
import renderer from 'react-test-renderer';

import App from '../App';

it('renders without crashing', () => {
  expect(renderer.create(<App />).toJSON()).toMatchSnapshot();
});
