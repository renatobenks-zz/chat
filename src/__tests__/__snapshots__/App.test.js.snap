// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<div
  className="App"
>
  <header
    className="App-header"
  >
    <img
      alt="logo"
      className="App-logo"
      src="test-file-stub"
    />
    <h1
      className="App-title"
    >
      Welcome to React
    </h1>
  </header>
  <p
    className="App-intro"
  >
    To get started, edit 
    <code>
      src/App.js
    </code>
     and save to reload.
  </p>
</div>
`;
